src/stringProcessing/checkNofollow.js

```diff
@@ -1,17 +1,39 @@
 /** @module stringProcessing/checkNofollow */
 
+// We use an external library, which can be found here: https://github.com/fb55/htmlparser2.
+let htmlparser = require( "htmlparser2" );
+let isString = require( "lodash/isString" );
+
 /**
- * Checks if a links has a nofollow attribute. If it has, returns Nofollow, otherwise Dofollow.
+ * Checks if a links has a nofollow attribute value. If it has, returns Nofollow, otherwise Dofollow.
  *
- * @param {string} text The text to check against.
+ * @param {string} anchorHTML The anchor HTML to check against.
  * @returns {string} Returns Dofollow or Nofollow.
  */
-module.exports = function( text ) {
-	var linkFollow = "Dofollow";
+module.exports = function( anchorHTML ) {
+	let linkFollow = "Dofollow";
+
+	let parser = new htmlparser.Parser( {
+		/**
+		 * Detects if there is a `nofollow` argument value in the `rel` argument of a link.
+		 *
+		 * @param {string} tagName The tag name.
+		 * @param {object} nodeValue The attribute with the keys and values of the tag.
+		 * @returns {void}
+		 */
+		onopentag: function( tagName, nodeValue ) {
+			if ( tagName !== "a" || ! isString( nodeValue.rel ) ) {
```

Comment:
I know you are trying to be explicit, but I think `tagName !== "a" || ! nodeValue.rel` should suffice to reduce the number of operations needed. HTML Attributes are always strings as far as I know.